api-server/config/openapi.yml

```diff
@@ -298,6 +298,69 @@ paths:
             application/json:
               schema:
                 $ref: '#/components/schemas/BrokersResponse'
+  /readAddressAttributes:
+    get:
+      summary: read address attributes
+      description: >
+        **Read values of address attributes**
+          The return value is a json array that contains
+          values of requested attributes of the addresses's mbean.
+
+          **Example:**
+          
+          **Request url:** https://localhost:9443/api/v1/readAddressAttributes?name=DLQ,attrs=RoutingTypes
+          (To read the `RoutingTypes` attribute of the address DLQ)
+
+          **Response:**
+          ```json
+          [
+            {
+              "request": {
+                "mbean": "org.apache.activemq.artemis:address=\"DLQ\",broker=\"amq-broker\",component=addresses",
+                "attribute": "RoutingTypes",
+                "type": "read"
+              },
+              "value": [
+                "ANYCAST"
+              ],
+              "timestamp": 1715864988,
+              "status": 200
+            }
+          ]
+          ```
+          **Note**: to read multiple attributes, set it to **attrs** parameter
+          separated by commas, e.g. `RoutingTypes,Address`.
+      tags:
+        - jolokia
+      operationId: readAddressAttributes
+      parameters:
+        - in: header
+          name: jolokia-session-id
+          schema:
+            type: string
+          required: true
+        - name: name
+          schema:
+            type: string
+          required: true
+          in: query
+        - name: attrs
+          description: attribute names separated by commas. If not speified read all attributes.
+          required: false
+          in: query
+          schema:
+            type: array
+            items:
+              type: string
```

Comment:
Can you provide the list of valid attributes in the specification as an enumeration please?
This will help:
* to know in the frontend what are the valid options to request
* to limit the scope of the endpoint to values we have control on

It should look something like this:
```
attrs:
  type: array
  description: attribute names separated by commas. If not specified read all attributes.
  items:
    type: string
    enum:
      - RoutingTypes
```

Reply:
@lavocatt Hi, this information can be dynamically discovered by a query to the api server, e.g.
`https://localhost:9443/api/v1/addressDetails?name=DLQ`
It will return all the operations and attributes available on address `DLQ`.

Reply:
The question would be, how often are we expecting this list to change?

Is it truly dynamic in the sense that it is defined by the runtime, or is it somehow a static content that is served every time by the jolokia endpoint?

If I remember correctly, the last time I called `addressDetails` on a couple of addresses, the returned object was always the same. My call would be that if we can expect this object to always be the same, then we should treat it as API definition.

There are multiple benefits of having a limited set of possibilities, the main of which is to keep the scope under control. The second by extension is to limit any abusive use by the user.

Reply:
For each compnent the operations and attributes are defined in the broker, for example here is the definition for address component https://github.com/apache/activemq-artemis/blob/2.33.0/artemis-core-client/src/main/java/org/apache/activemq/artemis/api/core/management/AddressControl.java
(it's documented in the api.md https://github.com/artemiscloud/activemq-artemis-self-provisioning-plugin/blob/main/api.md#getaddressdetails)

The definition may change over time but mostly the change is additive, i.e. new ops or attrs may be added in future releases but existing ones won't normally be changed.

Access control is a rather larger topic and not fit into this PR. First the console itself has a login, our plugin may adopt that, or have our own authorization framework. Second the jolokia access can be configured on the broker side. Here: https://github.com/apache/activemq-artemis/blob/2.33.0/docs/user-manual/management.adoc#configuring-jmx